app/controllers/vacation_requests_controller.rb

```diff
@@ -29,28 +29,36 @@ def update
     end
   end
 
+  def requested
+    requests = VacationRequest.where(user_id: params[:user_id]).requested
+    render json: requests
+  end
+
 private
 
   def set_vacation_request
     @vacation_request = VacationRequest.find params[:id]
   end
 
-  # Convert JS Date/Time in miliseconds to Ruby Date format
-  def date_from_ms(ms)
-    Time.zone.at(ms / 1000) unless ms.nil?
+  def vacation_request_params
+    prepare_date_params
+    params.require(:vacation_request)
+      .permit(:kind, :start, :end, :duration, :status)
   end
 
-  def prepare_date
-    date_in_ms = params[:vacation_request][:start]
-    params[:vacation_request][:start] = date_from_ms(date_in_ms) if date_in_ms
+  def vacation_params(val)
+    params[:vacation_request][val]
+  end
 
-    date_in_ms = params[:vacation_request][:end]
-    params[:vacation_request][:end]   = date_from_ms(date_in_ms) if date_in_ms
+  def prepare_date_params
+    date_from_ms(:start)
+    date_from_ms(:end)
   end
 
-  def vacation_request_params
-    prepare_date
-    params.require(:vacation_request)
-      .permit(:kind, :start, :end, :duration, :status)
+  # Convert JS Date/Time in miliseconds to Ruby Date format
+  def date_from_ms(val)
```

Comment:
Again according to method name it should return parsed data and not modifies params object.